superset-frontend/src/dashboard/util/replaceUndefinedByNull.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { cloneDeep } from 'lodash';
+
+export default function replaceUndefinedByNull(object: Object) {
```

Comment:
We run `cloneDeep` on every level of recursion - I think doing it once, in the first level of recursion, would be enough. However, I'd like to propose a different solution:
```
const replacer = (key, val) => {
  if (val === undefined) return null;
  return val;
}

const replaceUndefinedByNull = (object) => JSON.parse(JSON.stringify(object, replacer));
```
I'm not sure if it would be faster than your solution (we could time it to verify), but I think it's easier to understand. WDYT?

Reply:
I totally agree with the unnecessary cloneDeep. Will fix that. 

About the JSON serialization/deserialization I prefer the `deepClone` because it's faster. We already have benchmarks on this topic:
https://stackoverflow.com/questions/122102/what-is-the-most-efficient-way-to-deep-clone-an-object-in-javascript 
https://jsben.ch/bWfk9